pkg/sql/colexec/auto_incr.go

```diff
@@ -260,13 +260,40 @@ func updateBatchImpl(ColDefs []*plan.ColDef, bat *batch.Batch, offset, step []ui
 	return nil
 }
 
-func getCurrentIndex(param *AutoIncrParam, colName string) (uint64, uint64, error) {
-	rds, _ := param.rel.NewReader(param.ctx, 1, nil, nil)
-	for {
+func getCurrentIndex(param *AutoIncrParam, colName string, mp *mpool.MPool) (uint64, uint64, error) {
+	ctx := context.TODO()
+
+	var rds []engine.Reader
+
+	ret, err := param.rel.Ranges(ctx, nil)
+	if err != nil {
+		panic(err)
+	}
+	switch {
+	case len(ret) == 0:
+		rds, _ = param.rel.NewReader(ctx, 1, nil, nil)
```

Comment:
are you sure we can ignore the error here. please check again

Reply:
adjust to return err